lib/internal/crypto/scrypt.js

```diff
@@ -80,12 +80,25 @@ function check(password, salt, keylen, options, callback) {
 
   let { N, r, p, maxmem } = defaults;
   if (options && options !== defaults) {
-    if (options.hasOwnProperty('N'))
+    let has_N, has_r, has_p;
+    if (has_N = options.hasOwnProperty('N'))
```

Comment:
I know the `hasOwnProperty` has already been there, but we should use `undefined` as a measure of if an option was set, to be consistent with the rest of the API surface and to prevent `Object.create(null)` from breaking this.

Reply:
@TimothyGu Done!